text_generation/causal_lm/cpp/continuous_batching/library/src/tests/scheduler.cpp

```diff
@@ -0,0 +1,157 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+#include "openvino/runtime/core.hpp"
+#include "sequence_group.hpp"
+#include "scheduler.hpp"
+
+
+
+TEST(TestScheduler, general_test) {
+    SchedulerConfig scheduler_config {
+        // batch size
+        .max_num_batched_tokens = 32,
+        // cache params
+        .num_kv_blocks = 6,
+        .block_size = 4,
+        // mode - vLLM or dynamic_split_fuse
+        .dynamic_split_fuse = false,
+        // vLLM specific params
+        .max_num_seqs = 5,
+        .max_paddings = 8,
+    };
+    std::vector<uint64_t> tokens = {0,1,2,3,4,5,6,7};
+    SequenceGroup::Ptr sequence_group1 = std::make_shared<SequenceGroup>(0, ov::Tensor(ov::element::i64, {tokens.size()}, tokens.data()),
+                                                                            GenerationConfig::greedy(), scheduler_config.block_size);
+    SequenceGroup::Ptr sequence_group2 = std::make_shared<SequenceGroup>(1, ov::Tensor(ov::element::i64, {tokens.size()}, tokens.data()),
+                                                                            GenerationConfig::greedy(), scheduler_config.block_size);
+    std::vector<SequenceGroup::Ptr> requests = {sequence_group1, sequence_group2};
+                                                                       
+    
+    // schedule 2 sequence groups that use 4 kv blocks 
+    Scheduler scheduler = Scheduler(scheduler_config);
+    auto out1 = scheduler.schedule(requests);
+
+    std::vector<size_t> ref_ids = {0, 1};
+    EXPECT_EQ(out1.m_scheduled_sequence_groups_ids, ref_ids);
+    EXPECT_EQ(out1.m_block_tables[0].size(), 2);
+    EXPECT_EQ(out1.m_block_tables[1].size(), 2);
+    EXPECT_EQ(out1.m_block_tables[0][0]->is_free(), false);
+    EXPECT_EQ(out1.m_block_tables[0][0]->get_index(), 0);
+    EXPECT_EQ(out1.m_block_tables[0][1]->is_free(), false);
+    EXPECT_EQ(out1.m_block_tables[0][1]->get_index(), 1);
+    EXPECT_EQ(out1.m_block_tables[1][0]->is_free(), false);
+    EXPECT_EQ(out1.m_block_tables[1][0]->get_index(), 2);
+    EXPECT_EQ(out1.m_block_tables[1][1]->is_free(), false);
+    EXPECT_EQ(out1.m_block_tables[1][1]->get_index(), 3);
+    // tokens.size() * 2 tockens should be scheduled on promt phase, corresponding to first two sequences 
+    EXPECT_EQ(out1.m_total_num_scheduled_tokens, tokens.size() * 2);
+    EXPECT_EQ(out1.is_prompt, true);
+    for (auto seq: requests) {
```

Comment:
should we append new tokens to sequences to emulate _generation_  ?

Reply:
Done.